modules/javafx.graphics/src/main/native-font/directwrite.cpp

```diff
@@ -845,30 +845,35 @@ jobject newD2D1_MATRIX_3X2_F(JNIEnv *env, D2D1_MATRIX_3X2_F *lpStruct)
 /*                                                                        */
 /**************************************************************************/
 
-JNIEXPORT jlong JNICALL OS_NATIVE(_1WICCreateImagingFactory)
-    (JNIEnv *env, jclass that)
+JNIEXPORT jboolean JNICALL OS_NATIVE(CoInitializeEx)
+    (JNIEnv *env, jclass that, jint arg0)
 {
-    /* This routine initialize COM in order to create an WICImagingFactory.
-     * It runs on the prism thread and expects no other codes in this thread
-     * to interface with COM.
-     * Note: This method is called by DWFactory a single time.
-     */
-    HRESULT hr = CoInitializeEx(NULL, COINIT_APARTMENTTHREADED | COINIT_DISABLE_OLE1DDE);
+    HRESULT hr = CoInitializeEx(NULL, (DWORD)arg0);
 
-    /* This means COM has been initialize with a different concurrency model.
+    /* This means COM has been initialized with a different concurrency model.
      * This should never happen. */
-    if (hr == RPC_E_CHANGED_MODE) return NULL;
+    if (hr == RPC_E_CHANGED_MODE) return JNI_FALSE;
 
+    return SUCCEEDED(hr) ? JNI_TRUE : JNI_FALSE;
```

Comment:
As a follow-on, changing this line to the following fixes it for me:

```
     return JNI_TRUE;
```

Reply:
> The existing logic returns an error only when CoInitializeEx returns RPC_E_CHANGED_MODE. With this change, you will also return an error when CoInitializeEx returns S_FALSE (meaning COM is already initialized). It looks like this is why it is failing for me.

Strange, [MSDN](https://docs.microsoft.com/en-us/windows/desktop/api/winerror/nf-winerror-succeeded) claims that `SUCCEEDED` is implemented as
```
#define SUCCEEDED(hr) (((HRESULT)(hr)) >= 0)
```
and `S_FALSE` has the [value](https://docs.microsoft.com/en-us/windows/desktop/learnwin32/error-handling-in-com) of `0x1`.

Reply:
I just checked the docs, and I see that you are right. In that case, it must have been some other condition on my system that was causing it to report a failure (when I saw the failure I hadn't instrumented the code to print the return value, so I can't know for sure). It might be safer to return JNI_TRUE unconditionally, since that preserves the existing logic (I'm just guessing, but there must have been some reason that the original code only checked for `== RPC_E_CHANGED_MODE` rather than `!SUCCEEDED`).